<launch>

  <arg name="port" value="/dev/ttyUSB0" />
  <arg name="q_name" value="hummingbird_2" />
  <arg name="q_frame" value="/vicon/hummingbird_2/hummingbird_2" />
  <arg name="ugv_frame" value="/vicon/jackal1/jackal1" />
  <arg name="box_frame" value="/vicon/box/box" />
  <arg name="w_frame" value="/odom" />
  <arg name="tuning_file" value="default_h.yaml" />  

  <!-- Vicon connection node -->
  <include file="$(find vicon_bridge)/launch/vicon.launch" />

  <!-- Joy Controller xbox 360 -->
  <node name="joy_controller" pkg="joy" type="joy_node" output="screen"/>

  <!-- Asctec TRPY Controller -->
  <node pkg="asctec_ll_updated" name="$(arg q_name)" type="asctec_ll_updated_node" output="screen" respawn="false">
     <param name="port" value="$(arg port)"/>
     <param name="ll_status_rate" value="1"/>
     <param name="imu_calc_data_rate" value="0"/>
     <param name="rc_data_rate" value="0"/>
     <param name="ctrl_out_rate" value="10"/>
     <param name="gps_data_advanced_rate" value="0"/>
  </node>

  <!-- Asctec Position Controller -->
  <node name="asctec_controller" 
	pkg="pc_asctec_sim" 
	type="controller_node" 
	output="screen">

     <rosparam file="$(find pc_asctec_sim)/cfg/$(arg tuning_file)" command="load"/>

     <param name="q_name" value="$(arg q_name)"/>
     <param name="w_frame" value="$(arg w_frame)"/>
     <param name="q_frame" value="$(arg q_frame)"/>

  </node>

  <!-- Trajectory Generator -->
  <node name="waypoint_server" 
        pkg="pc_asctec_sim" 
        type="trajectory_real_node">

     <param name="q_name" value="$(arg q_name)"/>
     <param name="w_frame" value="$(arg w_frame)"/>
  </node>

  <!-- Mirror Hacker -->
  <node name="mir_hacker" 
        pkg="pc_asctec_sim" 
        type="mir_hack" 
        output="screen">

     <param name="q_name" value="$(arg q_name)"/>
     <param name="q_frame" value="$(arg q_frame)"/>
     <param name="box_frame" value="$(arg box_frame)"/>
     <param name="ugv_frame" value="$(arg ugv_frame)"/>
     <param name="w_frame" value="$(arg w_frame)"/>
  </node>

</launch>
